### PR TITLE
port pdfhummus to windows

### PR DESCRIPTION
**Pull request by jingkaimori**
* Before adding new features and new modules, please go to issues to submit the relevant feature description first.
* Write good commit messages and use the same coding conventions as the rest of the project.
* Please commit code to dev branch and we will merge into master branch in feature
* Ensure your edited codes with four spaces instead of TAB.

------

* 增加新特性和新模块之前，请先到issues提交相关特性说明，经过讨论评估确认后，再进行相应的代码提交，避免做无用工作。
* 编写友好可读的提交信息，并使用与工程代码相同的代码规范，代码请用4个空格字符代替tab缩进。
* 请提交代码到dev分支，如果通过，我们会在特定时间合并到master分支上。
* 为了规范化提交日志的格式，commit消息，不要用中文，请用英文描述。

### CODE REVIEW

File: packages/p/pdfhummus/xmake.lua [hunk 1]
```diff
@@ -36,14 +38,24 @@ package("pdfhummus")
...
                 add_files("PDFWriter/*.cpp")
                 add_headerfiles("(PDFWriter/*.h)")
                 add_packages("freetype")
                 add_packages("libtiff", "libpng", "libjpeg")
-                add_packages("libaesgm", "zlib")
+                add_packages("zlib")
+                if is_plat("windows","mingw") then
+                    add_packages("zlib")
+                    add_files("LibAesgm/*.c")
+                    add_includedirs("LibAesgm")
```
Comment (waruqi) [inline packages/p/pdfhummus/xmake.lua:55]: why not improve libaesgm package? it is from `pdfhummus/LibAesgm`.

Comment (jingkaimori) [reply packages/p/pdfhummus/xmake.lua:55]: libaesgm is imported by pdfhummus at 2017, but its first release is earlier than 2009. It's hard to determine newer libaesgm can be compiled on windows.

Comment (waruqi) [reply packages/p/pdfhummus/xmake.lua:55]: But now the libaesgm is exactly the same as the pdfhummus/LibAesgm, in addition, even if it does not compile on windows. We should also improve the package code of libaesgm. You can make libaesgm support Windows with add_patches

**Comment by jingkaimori** [comment]
why dynamic linking on windows fails?

**Comment by waruqi** [comment]
> why dynamic linking on windows fails?

it did not export c++ class symbols in shared library.

you can try adding this code in ported xmake.lua

```lua
if is_plat("windows") and is_kind("shared") then
    add_rules("utils.symbols.export_all", {export_classes = true})
end
```

**Comment by jingkaimori** [comment]
test script try to install libaesgm on windows|arm, even if pdfhummus is not supported on this platform. Is it a bug?

**Comment by waruqi** [comment]
> test script try to install libaesgm on windows|arm, even if pdfhummus is not supported on this platform. Is it a bug?

you need not disable window arm, It is experimental. I will merge it.

**Pull request merged**